src/clj/puppetlabs/services/jruby/jruby_puppet_core.clj

```diff
@@ -192,140 +167,25 @@
       (update-in [:master-run-dir] #(or % default-master-run-dir))
       (update-in [:master-log-dir] #(or % default-master-log-dir))
       (update-in [:max-active-instances] #(or % (default-pool-size (ks/num-cpus))))
```

Comment:
I think we want to take out the calls to provide default for `compile-mode`, `borrow-timeout`, and `max-active-instances`, and `max-borrows-per-instance` - AFAIK we're using the defaults in jruby-utils and not currently providing our own.